provider/spark_common.go

```diff
@@ -0,0 +1,221 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright 2024 FeatureForm Inc.
+//
+
+package provider
+
+import (
+	"github.com/aws/aws-sdk-go-v2/service/glue"
+	"github.com/featureform/config"
+	"github.com/featureform/fferr"
+	"github.com/featureform/filestore"
+	"github.com/featureform/helpers"
+	"github.com/featureform/logging"
+	"github.com/featureform/metadata"
+	pl "github.com/featureform/provider/location"
+	pc "github.com/featureform/provider/provider_config"
+	"github.com/featureform/provider/types"
+)
+
+func genericSparkSubmitArgs(execType pc.SparkExecutorType, deployMode types.SparkDeployMode, tfType TransformationType, outputLocation pl.Location, code string, sourceList []pysparkSourceInfo, jobType JobType, store SparkFileStoreV2, mappings []SourceMapping) (*sparkCommand, error) {
+	logger := logger.With("deployMode", deployMode, "outputLocation", outputLocation, "outputLocationType", fmt.Sprintf("%T", outputLocation), "code", code, "sourceList", sourceList, "jobType", jobType, "store", store)
+	logger.Debugw("SparkSubmitArgs")
+	snowflakeConfig, err := getSnowflakeConfigFromSourceMapping(mappings)
+	if err != nil {
+		logger.Errorw("Could not get Snowflake config from source mapping", "error", err)
+		return nil, err
+	}
```

Comment:
I may be getting ahead of myself here, but do we want to pull this kind of logic out into a separate function that can handle not only this case, but also BigQuery sources/configs/etc., too?